test/specs/oas-relative-servers/v3-relative-servers.spec.js

```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+const SwaggerParser = require("../../../lib");
+const fetch = require("node-fetch");
+const {expect} = require("chai");
+
+const RELATIVE_SERVERS_OAS3_URL = "https://petstore3.swagger.io/api/v3/openapi.json"; //Petstore v3 json has relative path in "servers"
```

Comment:
Thank you for this PR, glad I'm getting a chance to review it now! 

I would prefer to avoid adding dependencies to random URLs, as it can have impacts on stability of the test suite (already not good), and can make things hard for those on unstable or highly locked down environments.

Could you create the smallest replicatable file and save it in there, like `test/specs/deep-circular/deep-circular.yaml` has done. Then this test will not need to go over the wire.

Reply:
Hey Phil @philsturgeon , thanks much for the review comment. I'll have to mock the fetch from an https URL to get the proposed changes tested without going over the internet. Let me add that and push a commit by tomorrow.

Reply:
Hey Phil @philsturgeon , please take a look at the latest push. I had to resolve conflicts as the upstream repo had new changes on 08/15. So, there are some merge conflicts that I had to solve. I was able to mock the http url fetch and even added more test cases. Let me know your review comments. Thanks!